cardano-explorer-db/app/Explorer/App/DB/UtxoSet.hs

```diff
@@ -0,0 +1,104 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Explorer.App.DB.UtxoSet
+  ( utxoSetAtBlock
+  ) where
+
+import           Cardano.Chain.Common (fromCBORTextAddress, isRedeemAddress)
+import           Control.Monad (when)
+
+import           Data.Word (Word64)
+
+import           Explorer.DB
+
+import qualified Data.List as List
+import qualified Data.Map.Strict as Map
+import           Data.Text (Text)
+import qualified Data.Text as Text
+import qualified Data.Text.IO as Text
+
+import           System.IO (IOMode (..), withFile)
+
+utxoSetAtBlock :: Word64 -> IO ()
+utxoSetAtBlock slotNo = do
+  genesisSupply <- runDbNoLogging queryGenesisSupply
+  utxoSet <- runDbNoLogging $ queryUtxoAtSlotNo slotNo
+  fees <- runDbNoLogging $ queryFeesUpToSlotNo slotNo
```

Comment:
what about doing all 3 queries in a single transaction?
`(genesisSupply, utxoSet, fees) <- runDbNoLogging $ do`

Reply:
Sure, its possible, but I think anyone would be hard pressed to show one is better than the other.

Reply:
only problem i can foresee, is what if a rollback + rollforward happens between the 2 queries?

Reply:
This command would only ever be run on the stable (ie more than 2160 slots in the past) part of the chain.